src/apify_client/clients/resource_clients/webhook.py

```diff
@@ -0,0 +1,130 @@
+from typing import Any, Dict, List, Optional
+
+from ..base.resource_client import ResourceClient
+from .webhook_dispatch_collection import WebhookDispatchCollectionClient
+
+
+def prepare_webhook_representation(
+    event_types: Optional[List] = None,
+    request_url: Optional[str] = None,
+    payload_template: Optional[str] = None,
+    actor_id: Optional[str] = None,
+    actor_task_id: Optional[str] = None,
+    actor_run_id: Optional[str] = None,
+    ignore_ssl_errors: Optional[bool] = None,
+    do_not_retry: Optional[bool] = None,
+    idempotency_key: Optional[str] = None,
+    is_ad_hoc: Optional[bool] = None,
+) -> Dict:
+    """Prepare webhook dictionary representation for clients."""
+    webhook: Dict[str, Any] = {}
+
+    if event_types is not None:
+        webhook['eventTypes'] = event_types
+
+    if request_url is not None:
+        webhook['requestUrl'] = request_url
+
+    if payload_template is not None:
+        webhook['payloadTemplate'] = payload_template
+
+    condition = {}
+    if actor_run_id is not None:
+        condition['actorRunId'] = actor_run_id
+        webhook['isAdHoc'] = True
+    elif actor_task_id is not None:
+        condition['actorTaskId'] = actor_task_id
+    elif actor_id is not None:
+        condition['actorId'] = actor_id
+
+    if condition != {}:
+        webhook['condition'] = condition
+
+    if ignore_ssl_errors is not None:
+        webhook['ignoreSslErrors'] = ignore_ssl_errors
+
+    if do_not_retry is not None:
+        webhook['doNotRetry'] = do_not_retry
+
+    if idempotency_key is not None:
+        webhook['idempotencyKey'] = idempotency_key
+
+    if is_ad_hoc is not None:
+        webhook['isAdHoc'] = is_ad_hoc
+
+    return webhook
```

Comment:
Look at how I did something similar in https://github.com/apify/apify-client-python/pull/29/files. Basically, I introduced function that converts snake case to camel case. And then, when preparing these args, you can do something like

```
const new_dict = {}
for key, value of locals().entries():
    ...some custom logic you might have...
    new_dict[snake_case_to_camel_case(key)] = value;
```

It's much less verbose and easier to read imho.

Also, in you case, I'd handle `condition` independently of this loop, construct it explicitly and then just assign to the new dict.

Reply:
Good point.